test/object-store/sync/client_reset.cpp

```diff
@@ -1976,6 +1989,85 @@ TEST_CASE("sync: Client reset during async open", "[sync][pbs][client reset][baa
     after_callback_called.future.get();
 }
 
+TEST_CASE("sync: fast reconnect during client reset session suspend", "[sync][pbs][baas]") {
```

Comment:
does this test also hit the fix in `request_download_completion_notification()`?

Reply:
I haven't been able to create a test that reproduces hitting that assert with the original code. I'll revert the changes to that function.